[huangtian666] iP

huangtian666 opened the pull request:
# TalkGPT 🤖  

TalkGPT is your personal assistant, making conversations and managing tasks **smarter, faster, and more intuitive**.  

## Why TalkGPT?  
- 📝 **Text-based** and easy to use  
- ⚡ **Fast & responsive** conversations  
- 🎯 **Stores tasks effectively**  

## Getting Started 🚀  
1. **Download TalkGPT** from [here](https://github.com/huangtian666/ip).  
2. **Run the application**.
3. **Start chatting!** 💬  
4. Watch it **help you manage tasks, answer questions, and much more**.  

And yes, it's **FREE!** 🎉  

## Features  
- ✅ **To-do / Deadline / Event Manager**  
- ⏳ Smart reminders *(coming soon)*  
- 📅 Calendar & task management *(coming soon)* 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

huangtian666 committed:
renamed chatbot and added initial skeleton

huangtian666 committed:
improved skeletal version

huangtian666 committed:
added to do list

huangtian666 committed:
added to do list

huangtian666 committed:
added mark and unmark features

huangtian666 committed:
updated deadline and event files

huangtian666 committed:
cleaned files

huangtian666 committed:
added delete feature

huangtian666 committed:
cleaned main file and added error handling

huangtian666 committed:
fixed some logic in main file

huangtian666 committed:
cleaned files

huangtian666 committed:
added testing

huangtian666 committed:
updated delete

huangtian666 committed:
Add storage function

huangtian666 committed:
Revert "Add storage function"

This reverts commit f48932deba40d26e316d93a7d7e0341c79f8162a.

huangtian666 committed:
Reapply "Add storage function"

This reverts commit a69e790a206c7310131e005bd18c0aa90685f876.

huangtian666 committed:
Fix logic of mark tasks

huangtian666 committed:
Add features to recognise basic date format

huangtian666 committed:
Merge branch 'branch-Level-8'

huangtian666 committed:
Ensure a more OOP implementation

huangtian666 committed:
Split into packages

huangtian666 committed:
Clear files

huangtian666 committed:
Comply to coding standard

huangtian666 committed:
Follow coding standard

huangtian666 committed:
Fix welcome statement

huangtian666 committed:
Merge branch 'add-gradle-support'

huangtian666 committed:
Add test files

huangtian666 committed:
Fix gradle.build file

huangtian666 committed:
Add JavaDoc

huangtian666 committed:
Add find function to branch 9

huangtian666 committed:
Merge branch 'branch-Level-9'

huangtian666 committed:
Fix syntax error

huangtian666 committed:
Follow Coding Standard

huangtian666 committed:
Transfer all UI components to UI file

huangtian666 committed:
Set up JavaFX

huangtian666 committed:
Set up JavaFX

huangtian666 committed:
Set up interactive user interface

Mil-leon commented on src/main/java/talkgpt/command/Command.java:
```diff
@@ -0,0 +1,49 @@
+ *
+ * <p>Concrete subclasses include:</p>
+ * <ul>
+ *     <li>{@code ToDoCommand} - Adds a To-Do task.</li>
+ *     <li>{@code DeadlineCommand} - Adds a Deadline task.</li>
+ *     <li>{@code EventCommand} - Adds an Event task.</li>
+ *     <li>{@code DeleteCommand} - Deletes a task.</li>
+ *     <li>{@code ListCommand} - Lists all tasks.</li>
+ *     <li>{@code ListOnCommand} - Lists all tasks due a on specific date.</li>
+ *     <li>{@code MarkCommand} - Marks a task as completed or unmark a task.</li>
+ *     <li>{@code ExitCommand} - Exits the application.</li>
+ *     <li>{@code NextCommand} - Go to the next iteration of loop to read user input.</li>
+ *     <li>{@code HelpCommand} - Print all available commands to user</li>
+ *     <li>{@code ClearCommand} - Clear all tasks in the list</li>
```
Really like how you explain what each command does here 😄

Mil-leon commented on src/main/java/talkgpt/TaskList.java:
```diff
@@ -0,0 +1,190 @@
+import java.util.ArrayList;
+
+/**
+ * Manages the list of Task
+ * <p>
+ * This class handles various task operations such as add, delete, list and so on
+ * </p>
```
Some of the JavaDocs headers are missing punctuation. Majority of them do include punctuation

Mil-leon commented on src/main/java/talkgpt/parser/Parser.java:
```diff
@@ -0,0 +1,119 @@
+            ui.showMessage(Messages.Error.INVALID_INSTRUCTION.get());
+            return new NextCommand();
+        }
+    }
```
Perhaps switch/case can be used to make the code tidier? Nested conditionals make it a bit difficult to read

huangtian666 replied:
Thank you for your suggestion! I will change to switch statements 😊

Mil-leon commented on src/main/java/talkgpt/task/Event.java:
```diff
@@ -0,0 +1,123 @@
+     * @param description The event description.
+     * @param start       The event start time in the format "d/M/yyyy HHmm".
+     * @param end         The event end time in the format "d/M/yyyy HHmm".
+     */
```
Really like how well formatted your JavaDoc comment's are

Mil-leon commented on src/main/java/talkgpt/ui/Messages.java:
```diff
@@ -0,0 +1,117 @@
+     * </p>
+     */
+    public enum Error {
+        INVALID_DEADLINE("Invalid format! Use: deadline <description> /by <date>"),
+        INVALID_EVENT("Invalid format! Use: event <description> /from <start> /to <end>"),
+        EMPTY_TASK_LIST("You have no tasks yet!"),
+        INVALID_TASK_INDEX("Invalid task index! Please enter a valid number."),
+        DUPLICATE_TASK("talkgpt.task.Task already exists: "),
+        INVALID_DATE_FORMAT("Invalid date format! Please use dd/MM/yyyy."),
+        TASK_NOT_FOUND("No task found for the given date."),
+        NO_TASK_FOUND("No task found for the given word/sentence."),
+        INVALID_INSTRUCTION("Sorry, I don't understand ;(.");
```
I really like how you made the error messages an enum instead of individual messages 👍

Mil-leon reviewed (commented):
Overall really well done! Code was extremely well formatted and very easy to read, everything was organised well. Only a few minor touch ups should be necessary. :)

huangtian666 committed:
Fix launcher file

huangtian666 committed:
Anchor send button and text field

huangtian666 committed:
Link JavaFX to CSS

huangtian666 committed:
Improve UI

huangtian666 committed:
Adjust scroll bar

huangtian666 committed:
Adjust package

huangtian666 committed:
Add assertions for null and empty task descriptions

Added assertion statements to enforce task validity before adding it to the task list

These assertions prevent potential runtime errors by ensuring that:
- The task object is not null.
- The task description is not empty or whitespace.

This improves code reliability and helps catch logical issues early during development.

huangtian666 committed:
Fix code quality

huangtian666 committed:
Merge pull request #1 from huangtian666/branch-A-Assertions

Add assertions for null and empty task descriptions

huangtian666 committed:
Merge branch 'master' of https://github.com/huangtian666/ip

huangtian666 committed:
Merge pull request #2 from huangtian666/branch-A-CodeQuality

Fix code quality

huangtian666 committed:
Merge branch 'master' of https://github.com/huangtian666/ip

huangtian666 committed:
Set up CI

huangtian666 committed:
Change gradle.yml

huangtian666 committed:
Combine GUI with chatbot

huangtian666 committed:
Fix list_on feature logic

huangtian666 committed:
Improve GUI

huangtian666 committed:
Fix test files

huangtian666 committed:
Improve UI when exiting

huangtian666 committed:
Improve GUI for text input and button

huangtian666 committed:
Improve conversation

huangtian666 committed:
Add Ui.png

huangtian666 committed:
Update user guide

huangtian666 committed:
Change ReadME file

huangtian666 committed:
Update ReadME file

huangtian666 committed:
Delete the extra readme file

huangtian666 committed:
Improve readme

huangtian666 committed:
Increase the size of UI photo

huangtian666 committed:
Generate new jar file

huangtian666 committed:
Comply to code convention

huangtian666 committed:
Release version 1.0

huangtian666 committed:
Update ReadME

huangtian666 committed:
Fix error in delete task

huangtian666 committed:
Improve GUI of delete task

huangtian666 committed:
Update ReadME

huangtian666 committed:
Update Readme file

huangtian666 committed:
Update Readme

huangtian666 committed:
Add more error handling

huangtian666 committed:
Improve logic of duplicates handling

huangtian666 committed:
Correct error message

huangtian666 committed:
Change readme file

huangtian666 committed:
Change readme file
